runtime/doc/if_lua.txt

```diff
@@ -9,7 +9,109 @@ Lua Interface to Nvim					*lua* *Lua*
                                       Type <M-]> to see the table of contents.
 
 ==============================================================================
-1. Commands						*lua-commands*
+1. Importing modules					*lua-require*
+
+Neovim lua interface automatically adjusts `package.path` and `package.cpath` 
+according to effective &runtimepath value.  Adjustment happens after each time 
+'runtimepath' is changed, `package.path` and `package.cpath` are adjusted by 
+prepending directories from 'runtimepath' each suffixed by `/lua` and 
+`?`-containing suffixes from `package.path` and `package.cpath`. I.e. when 
+'runtimepath' option contains `/foo` and `package.path` contains only 
+`./a?d/j/g.nlua;./?.lua;/bar/?.lua` the resulting `package.path` after 
+adjustments will look like this: >
+
+	/foo/lua/?.lua;/foo/lua/a?d/j/g.nlua;./a?d/j/g.nlua;./?.lua;/bar/?.lua
```

Comment:
1. Why isn't there also `/foo/lua/bar/?.lua` ?
2. Is there any significance to the swapped of the order of `/foo/lua/?.lua;/foo/lua/a?d/j/g.nlua;`  vs the original order, `./a?d/j/g.nlua;./?.lua` ?